apps/etterlatte-saksbehandling-ui/client/src/components/behandling/soeknadsoversikt/familieforhold/barnepensjon/AnnenForelder.tsx

```diff
@@ -0,0 +1,128 @@
+import { FloppydiskIcon, PencilIcon, PersonIcon } from '@navikt/aksel-icons'
+import { BodyShort, Button, Heading, Radio, Textarea } from '@navikt/ds-react'
+import styled from 'styled-components'
+import { IconSize } from '~shared/types/Icon'
+import { AnnenForelder, Personopplysninger } from '~shared/types/grunnlag'
+import React, { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { ControlledRadioGruppe } from '~shared/components/radioGruppe/ControlledRadioGruppe'
+import { isPending } from '~shared/api/apiUtils'
+import { redigerAnnenForelder } from '~shared/api/behandling'
+import { useApiCall } from '~shared/hooks/useApiCall'
+import { formaterEnumTilLesbarString } from '~utils/formatering/formatering'
+
+const AnnenForelderBorder = styled.div`
+  padding: 1.2em 1em 1em 0em;
+  display: flex;
+`
+
+const IconWrapper = styled.span`
+  width: 2.5rem;
+`
+
+const PersonInfoWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 0.5rem;
+`
+
+type Props = {
+  behandlingId: string
+  personopplysninger: Personopplysninger
+}
+
+export const AnnenForelderBoks = ({ behandlingId, personopplysninger }: Props) => {
+  const [redigerModus, setRedigerModus] = useState<boolean>(false)
+  const [redigerStatus, redigerAnnenForelderRequest] = useApiCall(redigerAnnenForelder)
+
+  const lagreAnnenForelder = (annenForelder: AnnenForelder) => {
+    redigerAnnenForelderRequest(
+      {
+        behandlingId: behandlingId,
+        annenForelder: annenForelder,
+      },
+      () => {
+        setTimeout(() => window.location.reload(), 2000)
```

Comment:
refresh hele siden?

Reply:
Lagringen medfører oppdatering av grunnlag for behandlingen. Jeg hermer etter "rediger familierforhold" hvor det allerede var refreshing.

Reply:
hvis man bruker `mapResult()` for metoden som henter annen forelder så skal det være unødvendig å reloade. Ta tar react seg av re-renderingen

Reply:
Hele grunnlaget hentes på nytt fra PDL fordi en ny utgave av persongalleriet legges inn og vi må få behandlingen til å ta den i bruk. Egentlig kunne vi bare uppa grunnlagsversjonen når vi bare setter annenForelder, men vi har ikke noe støtte for det. Og diverse grunnlagsopplysninger kan brukes på diverse steder, så jeg føler det er tryggere å refreshe siden.